src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -48,42 +48,90 @@ class java_bytecode_convert_classt:public messaget
   {
   }
 
-  void operator()(const java_bytecode_parse_treet &parse_tree)
+  /// Converts a class parse tree into a class symbol and adds it to the
+  /// symbol table.
+  /// \param parse_trees: The parse trees found for the class to be converted.
+  /// \remarks
+  ///   Allows multiple definitions of the same class to appear on the
+  ///   classpath, so long as all but the first definition are marked with the
+  ///   attribute `@java::com.diffblue.OverlayClassImplementation`.
+  ///   Overlay class definitions can contain methods with the same signature
+  ///   as methods in the original class, so long as these are marked with the
+  ///   attribute @java::com.diffblue.OverlayMethodImplementation; such overlay
+  ///   methods are replaced in the original file with the version found in
+  ///   the last overlay on the classpath. Later definitions can also contain
+  ///   new supporting methods and fields that are merged in.
+  ///   This will allow insertion of Java methods into library classes to
+  ///   handle, for example, modelling dependency injection.
+  void operator()(
+    const java_class_loadert::parse_tree_with_overlayst &parse_trees)
   {
     // add array types to the symbol table
     add_array_types(symbol_table);
 
-    bool loading_success=parse_tree.loading_successful;
-    if(loading_success)
-      convert(parse_tree.parsed_class);
+    const java_bytecode_parse_treet &parse_tree = parse_trees.front();
+    const irep_idt &class_name = parse_tree.parsed_class.name;
+    bool loading_success = parse_tree.loading_successful;
+    // Check that the first class implementation isn't an overlay
+    if(is_overlay_class(parse_tree.parsed_class))
+    {
+      warning()
```

Comment:
OK so yeah the comment is correct - perhaps a task for a future day to allow the classes to come in any order